src/buttons/FAB.tsx

```diff
@@ -0,0 +1,141 @@
+import React from 'react';
+import { StyleSheet, Animated, StyleProp, ViewStyle } from 'react-native';
+import Button, { ButtonProps } from './Button';
+import { IconProps } from '../icons/Icon';
+import { FullTheme, withTheme } from '../config';
+
+export type FABProps = ButtonProps & {
+  iconName?: string;
+  iconColor?: string;
+  iconProps?: IconProps;
+  color?: string;
+  label?: string;
+  labelColor?: string;
+  size?: 'large' | 'small';
+  placement?: 'left' | 'right';
+  visible?: boolean;
+  disabled?: boolean;
+  upperCase?: boolean;
+  style?: StyleProp<ViewStyle>;
+  theme?: FullTheme;
+};
+
+const FAB: React.FunctionComponent<FABProps> = ({
+  iconName,
+  iconColor,
+  iconProps,
+  color,
+  label,
+  labelColor,
+  size = 'large',
+  visible = true,
+  disabled,
+  upperCase,
+  theme,
+  style,
+  titleStyle,
+  buttonStyle,
+  containerStyle,
+  iconContainerStyle,
+  placement,
+  ...props
+}) => {
+  const { current: animation } = React.useRef(
+    new Animated.Value(Number(visible))
+  );
+
+  React.useEffect(() => {
+    Animated.timing(animation, {
+      toValue: Number(visible),
+      duration: 200,
+      useNativeDriver: true,
+    }).start();
+  }, [animation, visible]);
+
+  return (
+    <Animated.View
+      style={[
+        {
+          opacity: animation,
+          transform: [{ scale: animation }],
+        },
+        styles.content,
+        placement && {
+          position: 'absolute',
+          margin: 16,
+          [placement]: 0,
+          bottom: 0,
+        },
+        style,
+      ]}
+    >
+      <Button
+        icon={{ name: iconName, color: iconColor || theme.colors.white }}
+        {...props}
```

Comment:
`{...props}` should always go at the bottom so that people can overwrite the defaults.